servers/rendering/shader_language.cpp

```diff
@@ -5585,6 +5585,12 @@ Error ShaderLanguage::_parse_block(BlockNode *p_block, const Map<StringName, Bui
 		} else if (tk.type == TK_CF_RETURN) {
 			//check return type
 			BlockNode *b = p_block;
+
+			if (b && b->parent_function && (b->parent_function->name == "vertex" || b->parent_function->name == "fragment" || b->parent_function->name == "light")) {
+				_set_error(vformat("'return' expression is forbidden in '%s' function!", b->parent_function->name));
```

Comment:
Maybe change it to `"'return' expression is forbidden in processor function '%s'!"`.

Reply:
Okay